project_close_forecast_date/models/purchase_order_line.py

```diff
@@ -0,0 +1,20 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from pytz import timezone
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    forecast_date = fields.Date(compute='_compute_forecast_date', store=True, readonly=True)
+
+    @api.depends('date_planned') 
+    def _compute_forecast_date(self):
+        for record in self:
+            record.forecast_date = record.get_date_tz(record.date_planned)
+
+    def get_date_tz(self, date):
+        tz = timezone(self.env.user.tz or self.company_id.partner_id.tz or 'UTC')

```

Comment:
At last, I've found this:

https://github.com/odoo/odoo/blob/6708408c9d2f8137271809962f0d39ee38a0a24b/odoo/fields.py#L1887-L1909

Such function could be called like .e.g. `fields.Date.context_today(self, timestamp=date)`, could you try?